drivers/regulator/regulator_shell.c

```diff
@@ -378,9 +378,9 @@ static int cmd_adset(const struct shell *sh, size_t argc, char **argv)
 		return -ENODEV;
 	}
 
-	if (strcmp(argv[2], "enable")) {
+	if (!strcmp(argv[2], "enable")) {
```

Comment:
because strcmp does not return a boolean, we should be comparing against 0, ie, `== 0`

Reply:
Done